lib/edamam_api_wrapper.rb

```diff
@@ -0,0 +1,47 @@
+require 'httparty'
+
+class EdamamApiWrapper
+  BASE_URL = "https://api.edamam.com/search"
+  APP_ID = ENV["APP_ID"]
+  APP_KEY = ENV["APP_KEY"]
+
+
+  @recipe_list = []
+
+  def self.search(query)
+    if !@recipe_list.empty?
+      @recipe_list = []
+    end
+
+    encoded_uri =  URI.encode("#{BASE_URL}?q=#{query}&app_id=#{APP_ID}&app_key=#{APP_KEY}&from=0&to=30")
+    response = HTTParty.get(encoded_uri)
+
+    if response ["hits"]
+      response["hits"].each do |recipe|
+        @recipe_list << Recipe.new(recipe["recipe"])
+      end
+    end
+    return @recipe_list
+  end
+
+  def self.show_recipe(uri)
+    recipe_uri_list = []
+
+    @recipe_list.each do |recipe|
+      recipe_uri_list << recipe.uri
+    end
+
+    if !recipe_uri_list.include?(uri)
```

Comment:
This is clever - I like the idea of not doing another search if the query is in the most recent results, and I'm glad that you figured out how to make it work if the query is not.

We could probably clean this up a little though:

```ruby
  def self.show_recipe(uri)
    # Check the most recent search results
    @recipe_list.each do |recipe|
      if recipe.uri == uri
        return recipe
      end
    end

    # Didn't find it, so make a new query
    recipe_uri_list = []
    # copy-paste lines 35-38, then end the method
  end
```